### PR TITLE
ALSA: hda - handle multiple i915 instances in acomp matching

### PR DESCRIPTION
**Pull request by kv2019i**
Currently i915_component_master_match() will return the first matching
i915 instance. This does not work in case system has multiple i915
instances. Extend the match logic to check that audio and display
controllers have a shared parent device.

### CODE REVIEW

**Comment by ranj063** [comment]
@kv2019i looks good to me. Which platform was this failing on?

File: sound/hda/hdac_i915.c [hunk 1]
```diff
@@ -76,8 +76,26 @@ EXPORT_SYMBOL_GPL(snd_hdac_i915_set_bclk);
...
+		return 0;
+
+	if (!strcmp(dev->driver->name, "i915") &&
+	    subcomponent == I915_COMPONENT_AUDIO &&
+	    PCI_SLOT(dev_pci->devfn) == PCI_SLOT(bus_pci->devfn))
+		match = 1;
+
+	dev_info(bus->dev, "i915 slot %u (%s), audio-dev slot %u (%s) match %d\n",
+		 PCI_SLOT(dev_pci->devfn), dev_name(dev),
+		 PCI_SLOT(bus_pci->devfn), dev_name(bus->dev), match);
```
Comment (lyakh) [inline sound/hda/hdac_i915.c]: do we really need a dev_info() for this? Looks like a dev_dbg() to me.

**Comment by kv2019i** [comment]
Thank you @juimonen and @lyakh for the reviews! This is still a draft for sharing and I put the dev_info()'s there on purponse. Once I get feedback from validation, I'll make fix this, add my signed-off and send upstream.

**Comment by plbossart** [comment]
@kv2019i Could you elaborate on the principle?

Audio is always 0000:00:1f.3 (domain 0, bus 0,  device 1f, function 3) - except for APL/GLK where it's ee:0 IIRC

What are the other cases we need to deal with for i915? I am not sure using a topology indicator is enough, we may have to tighten the criteria a bit, no?

**Comment by kv2019i** [comment]
This approach not sufficient. Closing the PR for now.

**Pull request closed without merging**